enterprise/internal/licensing/types.go

```diff
@@ -0,0 +1,15 @@
+package licensing
+
+type LicenseCheckRequestParams struct {
+	ClientSiteID string `json:"siteID"`
+}
+
+type LicenseCheckResponseData struct {
+	IsValid bool    `json:"is_valid"`
+	Reason  *string `json:"reason"`
+}
+
+type LicenseCheckResponse struct {
+	Error *string                   `json:"error"`
+	Data  *LicenseCheckResponseData `json:"data"`
+}
```

Comment:
If you want to make JSON fields optional, you can add `omitempty` like this:

```suggestion
package licensing

type LicenseCheckRequestParams struct {
	ClientSiteID string `json:"siteID"`
}

type LicenseCheckResponseData struct {
	IsValid bool    `json:"is_valid"`
	Reason  string `json:"reason,omitempty"`
}

type LicenseCheckResponse struct {
	Error string                   `json:"error,omitempty"`
	Data  *LicenseCheckResponseData `json:"data"`
}
```

Then you don't have to deal with string pointers, which can be annoying!

Reply:
TIL, thanks for the suggestion! Updated 191b270c9f7f7ac584f514332762502be173ae75.